bigtable-client-core/src/main/java/com/google/cloud/bigtable/grpc/io/ChannelPool.java

```diff
@@ -166,4 +165,62 @@ public int size() {
   public int availbleSize() {
     return channels.get().size();
   }
+
+  @Override
+  public synchronized ManagedChannel shutdown() {
+    for (ManagedChannel channel : channels.get()) {
+      channel.shutdown();
+    }
+    this.shutdown = true;
+    return this;
+  }
+
+  @Override
+  public boolean isShutdown() {
+    for (ManagedChannel channel : channels.get()) {
+      if (!channel.isShutdown()) {
+        return false;
+      }
+    }
+    return true;
+  }
+
+  @Override
+  public synchronized boolean isTerminated() {
+    List<ManagedChannel> c = new ArrayList<>(channels.get());
```

Comment:
I don't think there is a need to clean up the channels list. From the name, isTerminated shouldn't change the state of the object.